### PR TITLE
Feature/update typing

### PR DESCRIPTION
**Pull request by gimppa**
### Description
Updated type annotations all around. Mostly changed dict and list type annotations to be Dict[something] and List[something].

### Related issues

Fixes #789 github workflows

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Testing

- [x] Tests do not apply

### CODE REVIEW

**Comment by csc-felipe** [comment]
Why should we use `Dict` and `List` instead of `dict` and `list` in Python 3.11? Which makes me think that perhaps we [should set `python-version` for `mypy`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-python_version).

File: metadata_backend/api/handlers/rems_proxy.py [hunk 1]
```diff
@@ -75,3 +78,6 @@ def add_organization(organization_id: str, organization_name: str) -> None:
             organizations[org_id]["licenses"].append({"id": lic["id"], **title})
 
         return web.json_response(data=list(organizations.values()))
+
+
+# : Dict[str, Dict[str, Dict[str, Any] | Any] | Any]
```
Comment (csc-jm) [inline metadata_backend/api/handlers/rems_proxy.py:83]: Not quite sure what this is

File: metadata_backend/api/handlers/restapi.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 """Handle HTTP methods for server."""
 import json
 from math import ceil
-from typing import AsyncIterator, Dict, Iterator, List, Optional, Tuple, Union
+from typing import Any, AsyncIterator, Dict, Iterator, List, Optional, Tuple
```
Comment (csc-jm) [inline metadata_backend/api/handlers/restapi.py:4]: So according to PEP 585, these would instead be
- dict
- list
- tuple
- collections.abc.AsyncIterator
- collections.abc.Iterator

However, `Any` and `Optional` would still be imported from `typing`

**Review by csc-jm — CHANGES_REQUESTED** [review]
I think what @csc-felipe commented is a good point. If we're updating the typing, we might as well do it with the current best practice while we're at it (as long as it's not an unnecessarily long task). So while the built-in `typing` class still exists, it's been deprecated by now as proposed in PEP 585. So you would have `list`, `dict` etc. or then `collections.abc.Callable` and `collections.abc.Iterator`. 

However, it seems the types `Any`, `Optional`, `TypedDict` and `TypeVar` which we still have in our code (at least I found in this PR) are still to be imported from `typing` 🤔

File: metadata_backend/api/handlers/rems_proxy.py (hunk 1 above)
Comment (gimppa) [reply metadata_backend/api/handlers/rems_proxy.py]: This is a mistake :laughing:

**Review by csc-felipe — APPROVED** [review]
Looks good :)

**Comment by csc-felipe** [comment]
Approved, but you still have to click on the **three dots > dismiss review** from Joonatan's requested changes.

**Review by csc-jm — APPROVED** [review]
Now everything looks good to me as well :P

**Pull request merged**